### ISSUE TITLE
[Support] Create support form

### ISSUE THREAD

**Issue by ianb** [mozilla/fxa#797]
We want a support form, which will ultimately be hosted on `https://accounts.firefox.com/premium-support` (we want to be clear this is not a general FxA or Firefox support forum).

The page should require authentication, and should require an active subscription (Q: are we sure about the active subscription?). External pages may link directly to the support page.

Input fields TBD.

**Comment by shane-tomlinson** [comment]
Will this page be a panel under /settings or it's own separate page? 

If under settings, let's make the URL consistent with other settings panels and use `https://accounts.firefox.com/settings/premium-support`.

**Comment by clouserw** [comment]
Support wireframe is on page 11 here: https://mozilla.invisionapp.com/share/K2RBDY3VU9X#/screens/356896706

**Comment by lmorchard** [comment]
We just touched on this in a review of PR #961: Should this support form actually live in fxa-payments-server rather than fxa-content-server?

**Comment by shane-tomlinson** [comment]
If this is within the content server, I can pick this up once I finish #1098.

**Comment by lmorchard** [comment]
Re: living on the content server - just to summarize [my comment](https://github.com/mozilla/fxa/issues/1100#issuecomment-492712305) from #1100 here: 

Probably the only wrinkle that suggests the support form should live on payments-server is that it's a) linked to from subscription management and b) the form returns to subscription management after submission.

**Comment by juliastarkov** [comment]
Ready for production support mockups can be viewed here (screens 6 - 10): https://mozilla.invisionapp.com/share/5BS1I7KANWP#/363599381_Support

**Comment by juliastarkov** [comment]
Ready for production Zeplin mockups can be viewed here: https://zpl.io/2pMEYRN

**Comment by chenba** [comment]
@juliastarkov I haven't used Zeplin before.  Do I need access?  If so, invite bchen@mozilla.com or Zeplin username bchenistaken.

**Comment by chenba** [comment]
Re: content server vs. payment server

The code on the content server simply navigate the user to the payment server for subscription management.  (Because of #1637, I didn't realize this until I look at the code for the subscriptions panel code in content server.)

Given that, and what the mockups are showing, I _think_ this form should be on the payment server?  It might be confusing to users if we navigate them to the payment server for subscription management, and bounce them back for the support form, and send them back to the payment server...?

In other words, I just understood and regurgitated @lmorchard's comment https://github.com/mozilla/fxa/issues/797#issuecomment-492713468

**Comment by chenba** [comment]
More questions (for... @johngruen?):

Since only users with subscriptions can use the form, should we not display the "Contact Support" button on the Subscriptions page if the user does not have a subscription?

Currently when a user successfully canceled a subscription, there's a link on the modal going to the support form.  But if the canceled subscription is the last one, then that'd be a problem, as the user now should not be able to use the form?

If someone without any subscriptions visits the support form directly (and they have a valid access token), should they be redirected back to the Subscriptions page?

**Comment by juliastarkov** [comment]
@chenba Replies in order:
1. If the user does not have a subscription they will not have subscription page access and will not see an option to manage their subscription on their accounts home page. 
2. When a user cancels their subscription, they will have access to support (and the product) until the last day of their billing cycle. Once the billing cycle is complete, it will revert to what I wrote in point 1.
3. I'm not sure how a valid access token works here, but if the user doesn't have an active subscription the link to the form should reveal an error message (and an option to redirect to subscription page).

**Comment by chenba** [comment]
@bbangert pointed out that the lifespan of the access token is rather short, so that could be an issue when the user spend a lot of time typing out a support request only to lose it because of an expired token, if the form lives on the payments server.

The current flow is to take the user back to the subscription management page on the payments server after a successful form submission.  A janky aspect about this, with the form on the content server, is the successful submission message: it'll need to live on the payments server.  (Yay monorepo!)

**Comment by chenba** [comment]
@juliastarkov in the mock-ups, there's a checkbox to opt in to allow Moz to collection device data.  Where can I find more info on that?  I don't know what it means.  Is there a similar feature I can took a look at?  Thanks!

**Comment by juliastarkov** [comment]
@jgruen Can you please see Barry's question above? I'm not sure where to find another example of a data collection check box in our products (if there is one).

**Comment by philbooth** [comment]
> there's a checkbox to opt in to allow Moz to collection device data. Where can I find more info on that? I don't know what it means. Is there a similar feature I can took a look at?

@chenba, we have checkbox styling in the content server for the choose-what-to-sync screen, if that's any help to you?

<img width="658" alt="Screenshot of the FxA choose-what-to-sync screen" src="https://user-images.githubusercontent.com/64367/60953202-39f74a00-a2ec-11e9-91bd-643314e464c4.png" />

The markup is in `app/scripts/templates/choose_what_to_sync.mustache` in the content server directory.

**Comment by johngruen** [comment]
@chenba, @juliastarkov I'm guessing barry's question here is what data are we trying to collect and/or pass along in the form...short answer is: I don't know. Why don't i sync up with Rina today and we can discuss on Slack

**Comment by chenba** [comment]
> I'm guessing barry's question here is what data are we trying to collect and/or pass along in the form

Yes.  Sorry that I was not clear.

**Comment by chenba** [comment]
@juliastarkov if the support form submission fails for some reason, what should that look like?

**Comment by juliastarkov** [comment]
@chenba It should be an error modal like this: http://mozilla.github.io/fxa/fxa-payments-server/?path=/story/routes-product-payment-failures--card-declined
code: https://github.com/mozilla/fxa/blob/master/packages/fxa-content-server/app/styles/modules/_modal.scss

There should be copy for the modal in the errors spreadsheet that was created by Jeff Pfaller. Let me know if you need access to that.

**Comment by johngruen** [comment]
Hey can we make the path here just be `/support`

**Comment by chenba** [comment]
> There should be copy for the modal in the errors spreadsheet that was created by Jeff Pfaller. Let me know if you need access to that.

@juliastarkov Yes, please.

**Comment by chenba** [comment]
> Hey can we make the path here just be `/settings/support` I don't really think we're using the word "premium" in public facing contexts

@johngruen Or how about just `/support`?

**Comment by johngruen** [comment]
wfm

**Comment by chenba** [comment]
Error message for failed form submission: "Hmm, we're having trouble with our system. We're working on fixing it for you and apologize for the inconvenience. Please try again later."

**Comment by chenba** [comment]
Related to #926 as there are a couple spots where we need to take the user back to the subscriptions management page.

**Comment by chenba** [comment]
Remove the "Subscriptions" heading on mobile.

**Comment by jgruen** [comment]
@juliastarkov keep up the good work, I'm dropping off this thread
